pkg/syncproto/sync_proto.go

```diff
@@ -102,13 +103,23 @@ type SerializedUpdate struct {
 	UpdateType api.UpdateType
 }
 
-func (s SerializedUpdate) ToUpdate() api.Update {
+var ErrBadKey = errors.New("Unable to parse key.")
+
+func (s SerializedUpdate) ToUpdate() (api.Update, error) {
 	// Parse the key.
-	parsedKey := model.KeyFromDefaultPath(string(s.Key))
-	parsedValue, err := model.ParseValue(parsedKey, s.Value)
-	if err != nil {
-		log.WithField("rawValue", string(s.Value)).Error(
-			"Failed to parse value.")
+	parsedKey := model.KeyFromDefaultPath(s.Key)
+	if parsedKey == nil {
+		log.WithField("key", s.Key).Error("BUG: cannot parse key.")
+		return api.Update{}, ErrBadKey
+	}
```

Comment:
I _think_ so.  If we blow up here then all the Typha's will die just because someone added a new piece of data to the datastore that we weren't expecting.  I think it's better to keep going in that case since we know there are keys (which felix doesn't need) that we can't handle here.